ci/infra/testrunner/tests/conftest.py

```diff
@@ -12,6 +23,97 @@ def pytest_addoption(parser):
+    >>>     c.run("kubectl ...")
+
+    Once 'c' goes out of scope the cluster directory from skuba will be removed and
+    the terraform deployment will be destroyed.
+    """
+    def __init__(self, conf: BaseConfig, masters=1, workers=1, platform='openstack'):
+        self.masters = masters
+        self.workers = workers
+        self.workspace = Workspace()
+        self.config = BaseConfig(conf.yaml_path)
+        self.__adjust_directories()
+        self.platform = platforms.get_platform(self.config, platform)
+        self.__copy_configuration(conf)
+        self.skuba = Skuba(self.config, platform)
+
+    def __enter__(self):
+        LOGGER.info("Provisioning cluster")
+        self.platform.provision(num_master=self.masters,
+                                num_worker=self.workers)
+        LOGGER.info("Initializing cluster")
+        self.skuba.cluster_init()
+        LOGGER.info("Bootstrap cluster")
+        self.skuba.node_bootstrap()
+        for i in range(self.masters):
+            LOGGER.info(f"Join master {i}")
+            self.skuba.node_join(role="master", nr=i)
+        for i in range(self.workers):
+            LOGGER.info(f"Join worker {i}")
+            self.skuba.node_join(role="worker", nr=i)
+        return super().__enter__()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        LOGGER.info("Cleanup cluster")
+        self.skuba.cleanup(self.config)
+        LOGGER.info("Cleanup terraform")
+        # Right now there is no additional check if the terraform cleanup
+        # worked, as it should throw an exception in case it fails breaking
+        # here anyhow
+        self.platform.cleanup()
+        LOGGER.info("Cleanup workspace")
+        self.workspace.cleanup()
+
+    def __adjust_directories(self):
+        """Reset the config workspace, to not create all cluster resources in the same
+        directory.
+
+        """
+        LOGGER.debug(f"Adjusting workspace to {self.workspace.temporary_directory}")
+        self.config.workspace = self.workspace.temporary_directory
+        LOGGER.debug(f"Adjusting terraform directory to {self.workspace.temporary_directory}")
+        self.config.terraform.tfdir = self.workspace.temporary_directory
+
+    def __copy_configuration(self, original_config: BaseConfig):
+        """Copy the terraform configuration files from the initial directory to the
+        newly generated ones.
+
+        """
+        dirname = os.path.basename(self.platform.tfdir)
+        src = f"{original_config.terraform.tfdir}/{dirname}"
+        tgt = f"{self.workspace.temporary_directory}/{dirname}"
+        LOGGER.info(f"Copying configuration: {src} --> {tgt}")
+        shutil.copytree(src, tgt)
+
+
+@pytest.fixture
+def cluster(conf: BaseConfig):
+    """Wrap the Cluster creation class in a function, so the tests don't have to
+    get the `conf` file.
+
+    """
+    def _(masters=1, workers=1, platform='openstack'):
```

Comment:
I think platform shouldn't have a default value and should always be passed from the test using the `target` fixture. This guarantees the test uses the platform defined in the environment.